reconcile/utils/terraform_client.py

```diff
@@ -531,34 +531,48 @@ def cleanup(self):
         for _, wd in self.working_dirs.items():
             shutil.rmtree(wd)
 
-    def _is_ignored_rds_modification(
+    def _get_pending_rds_modification(
         self, account_name: str, resource_name: str, resource_change: Mapping[str, Any]
-    ) -> bool:
+    ) -> list[str]:
         """
-        Determine whether the RDS resource changes are cases where a terraform apply
+        Determine whether the RDS resource changes are cases where a Terraform apply
         should be skipped.
         """
+        # A commonly changed RDS database settings that can be applied either
+        # immediately or during the next scheduled maintenance window, see:
+        #  https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.DBInstance.Modifying.html#USER_ModifyInstance.Settings
+        #
+        # Note that changing the RDS database identifier (the database name) is
+        # a complex operation that also involves Terraform performing a resource
+        # replacement and thus it's not included here, see:
+        #   https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_RenameInstance.html
+        supported_values: dict[str, str] = {
+            "engine_version": "EngineVersion",
+            "storage_type": "StorageType",
+            "allocated_storage": "AllocatedStorage",
+            "instance_class": "DBInstanceClass",
+            "multi_az": "MultiAZ",
+            "iops": "Iops",
+        }
         before = resource_change["before"]
         after = resource_change["after"]
-        changed_terraform_args = [
+        changed_resource_arguments = [
             key for key, value in before.items() if value != after.get(key)
         ]
-        if (
-            len(changed_terraform_args) == 1
-            and "engine_version" in changed_terraform_args
-            and self._aws_api is not None
-        ):
+        if len(changed_resource_arguments) > 0 and self._aws_api is not None:
```

Comment:
nit: you could rely on python truthyness
```suggestion
        if changed_resource_arguments and self._aws_api:
```

Reply:
Done. Thank you!